pkg/common/cns-lib/volume/manager.go

```diff
@@ -223,14 +235,14 @@ func (m *defaultManager) CreateVolume(ctx context.Context, spec *cnstypes.CnsVol
 		}
 		// Remove the taskInfo object associated with the volume name when the current task fails.
 		//  This is needed to ensure the sub-sequent create volume call from the external provisioner invokes Create Volume
-		taskDetailsInMap, ok := volumeTaskMap[spec.Name]
+		taskDetailsInMap, ok := volumeTaskMap[volNameFromInputSpec]
```

Comment:
Should we do the same at line 253 as well?

Reply:
Fixed.